usaspending_api/references/management/commands/load_naics.py

```diff
@@ -1,48 +1,74 @@
+import logging
+import os.path
+
+from django.conf import settings
 from django.core.management.base import BaseCommand
+from openpyxl import load_workbook
+from django.db import transaction
+import re
+import glob
+
 from usaspending_api.references.models import NAICS
-import os
-import csv
-import logging
-import django
 
 
 class Command(BaseCommand):
-    help = "Loads program information obtained from csv file on census.gov"
+    help = "Updates DB from Excel spreadsheets of USAspending terminology definitions into the naics model"
 
     logger = logging.getLogger('console')
 
-    DEFAULT_DIRECTORY = os.path.normpath('usaspending_api/references/management/commands/')
-    DEFAULT_FILEPATH = os.path.join(DEFAULT_DIRECTORY,  'naics_codes.csv')
+    path = 'usaspending_api/data/naics_archive'
+    path = os.path.normpath(path)
+    default_path = os.path.join(settings.BASE_DIR, path)
 
     def add_arguments(self, parser):
-            parser.add_argument(
-                '-f', '--file',
-                default=self.DEFAULT_FILEPATH,
-                help='path to CSV file to load',
-            )
+        parser.add_argument('-p', '--path', help='the path to the Excel spreadsheets to load', default=self.default_path)
+        parser.add_argument('-a', '--append', help='Append to existing guide', action='store_true', default=True)
 
     def handle(self, *args, **options):
+        load_naics(path=options['path'], append=options['append'])
+
+
+@transaction.atomic
+def load_naics(path, append):
+
+    if append:
+        logging.info('Appending definitions to existing guide')
+    else:
+        logging.info('Deleting existing definitions from guide')
+        NAICS.objects.all().delete()
+
+    # year regex object precompile
+    p_year = re.compile("(20[0-9]{2})")
+
+    dir_files = glob.glob(path + "/*.xlsx")
+
+    for fname in os.listdir(path):
```

Comment:
Delete lines 45-48 (line 43 finds all of the files).

The for loop on line 50 is looping over ever file twice

Reply:
redundant lines removed, for loop should be the only iteration through file